tests/e2e/ctl_v3_auth_no_proxy_test.go

```diff
@@ -34,3 +39,103 @@ func TestCtlV3AuthCertCNAndUsername(t *testing.T) {
 func TestCtlV3AuthCertCNAndUsernameNoPassword(t *testing.T) {
 	testCtl(t, authTestCertCNAndUsernameNoPassword, withCfg(*e2e.NewConfigClientTLSCertAuth()))
 }
+
+func TestCtlV3AuthCertCNWithWithConcurrentOperation(t *testing.T) {
+	e2e.BeforeTest(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// apply the certificate which has `root` CommonName,
+	// and reset the setting when the test case finishes.
+	t.Log("Apply certificate with root CommonName")
+	resetCert := applyTLSWithRootCommonName()
+	defer resetCert()
```

Comment:
There should be a separate cert bundle and an option in e2e tests pick certs with CommonName.

Reply:
I thought about enhancing the e2e framework, but it needs more time. I will think about it in a separate PR. Let's get 3.5.6 out firstly.

Reply:
Just rebased this PR, and added a TODO comment. PTAL.

Reply:
Sounds good.